ios/RNSPercentDrivenInteractiveTransition.mm

```diff
@@ -0,0 +1,69 @@
+#import "RNSPercentDrivenInteractiveTransition.h"
+
+@implementation RNSPercentDrivenInteractiveTransition {
+  RNSScreenStackAnimator *_animationController;
+}
+
+#pragma mark - UIViewControllerInteractiveTransitioning
+
+- (void)startInteractiveTransition:(id<UIViewControllerContextTransitioning>)transitionContext
+{
+  [super startInteractiveTransition:transitionContext];
+}
+
+#pragma mark - UIPercentDrivenInteractiveTransition
+
+// `updateInteractiveTransition`, `finishInteractiveTransition`,
+// `cancelInteractiveTransition` are forwared by superclass to
+// corresponding methods in transition context. In case
+// of "classical CA driven animaitons", such as UIView animation blocks
+// or direct utilization of CoreAnimation API, context drives the animation,
+// however in case of UIViewPropertyAnimator it does not. We need
+// to drive animation manually and this is exactly what happens below.
+
+- (void)updateInteractiveTransition:(CGFloat)percentComplete
+{
+  if (_animationController != nil) {
+    [self.animationController.inFlightAnimator setFractionComplete:percentComplete];
+  }
+  [super updateInteractiveTransition:percentComplete];
```

Comment:
Not sure why header looks so bad in those transitions, I think this line should control the state of animation of header too. Are those values not changing pretty linearly ?

Reply:
Header looks bad only in non-interactive transitions. TBH I'm surprised that it does not look bad in old implementation - the [docs suggest that it should look bad](https://github.com/software-mansion/react-native-screens/blob/960873aa03b9e22136d2f7b3b272e407011568d2/src/types.tsx#L395)

Reply:
I do not know what code drives the header transition in non-interactive transitions (notice that I'm not swiping on these recordings). Need to spend some time on this.

Reply:
The docs mention the native header transition where the back button title travels to the place of title when navigating back/forward. It works only with the default animations. For `simple_push` it was the fade animation in header iirc.

Reply:
I guess you're right. I've just confirmed that it has not been a fluke - on main there is nice fade transition. Do you happen to remember where this is animated? Or is this handled entirely by UIKit?

Reply:
Figured this out in: https://github.com/software-mansion/react-native-screens/pull/2477/commits/7a2dfc948eebc056700e9281d496cfc215a0a174